src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java

```diff
@@ -1094,8 +1095,12 @@ private User makeUser(User user, ExtImpRubicon rubiconImpExt) {
                 && liverampId == null
                 && resolvedId == null
                 && Objects.equals(userBuyeruid, resolvedBuyeruid)
-                && !hasStypeToRemove) {
-            return user;
+                && !hasStypeToRemove
+        ) {
+
+            return hasDataToRemove
+                    ? user.toBuilder().data(null).build()
+                    : user;
```

Comment:
I think no need to optimize this case
```
&& !hasStypeToRemove
&& !hasDataToRemove) {
    return user;
}
```

Reply:
There is a need because if we don't have ext, with checks in your approach, this ext will be created.